js/components/generate/generate.js

```diff
@@ -239,6 +253,12 @@ class GenerateForm extends React.Component {
   render() {
     return (
       <div style={{ display: "flex", flexDirection: "row", height: "100%" }}>
+        {this.state.showWarning && (
+          <ErrorMessage
+            message={`The courses [${this.state.invalidCourses}] were invalid! Please check your input.`}
```

Comment:
Have separate messages for singular vs. plural course counts. Note that in the singular form, "was invalid" should be used.